test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiXmlTests.cs

```diff
@@ -46,5 +46,35 @@ public void ParseBasicXmlShouldSucceed()
                     });
             }
         }
+
+        [Fact]
+        public void ParseBasicXmlWithBrokenUrlShouldSucceed()
+        {
+            using (var stream = Resources.GetStream(Path.Combine(SampleFolderPath, "basicXmlWithBrokenUrl.yaml")))
+            {
+                var yamlStream = new YamlStream();
+                yamlStream.Load(new StreamReader(stream));
+                var yamlNode = yamlStream.Documents.First().RootNode;
+
+                var context = new ParsingContext();
+                var diagnostic = new OpenApiDiagnostic();
+
+                var node = new MapNode(context, diagnostic, (YamlMappingNode)yamlNode);
+
+                // Act
+                var xml = OpenApiV3Deserializer.LoadXml(node);
+
+                // Assert
+                xml.ShouldBeEquivalentTo(
+                    new OpenApiXml
+                    {
+                        Name = "name1",
+                        // TODO: Verify that the URL caused a parsing error.
```

Comment:
Looks like a candidate for one more assertion for `diagnostic `.

Reply:
I changed this to a success case since that's what the spec allows (relative url should also work).

---
In reply to: [164986283](https://github.com/Microsoft/OpenAPI.NET/pull/197#discussion_r164986283) [](ancestors = 164986283)

Reply:
I created a new test case with empty URL. The diagnostic should contain an error for empty URL.

---
In reply to: [168329298](https://github.com/Microsoft/OpenAPI.NET/pull/197#discussion_r168329298) [](ancestors = 168329298,164986283)